src/client/Microsoft.Identity.Client/AppConfig/AssertionRequestOptions.cs

```diff
@@ -30,5 +31,11 @@ public class AssertionRequestOptions {
         /// Claims to be included in the client assertion
         /// </summary>
         public string Claims { get; set; }
+
+        /// <summary>
+        /// Capabilities that the client application declares in the assertion.
+        /// The service may require or interpret these capabilities in advanced scenarios.
+        /// </summary>
+        public IEnumerable<string> ClientCapabilities { get; set; }
```

Comment:
Not opposed to this, but what is the protocol? If the app is cp1 capable, then the assertion provider should also declare itself cp1 capable?

Reply:
short answer is yes, so that the ESTS sees a consistent set of capabilities. With MSI v1 Token Revocation we would also need capabilities.